integration/pubsub_test.go

```diff
@@ -547,35 +550,54 @@ func checkResultHandler(publishedMsg string, publishedMsgType string, ch chan<-
 	return message.HandleFunc(
 		func(ctx context.Context, msg *message.Message) message.Handler {
 			if publishedMsg != msg.Data() {
+				errHandler := message.Error(errors.New("published message and received message are different"))
```

Comment:
these changes are ensuring that we first run the Complete or Abandon handlers, before we push the result in the channel.
pushing the result in the channel will continue the processing in the test, which closes the connection, but Complete/Abandon send a frame on the amqp link. if we don't close the connection before, Complete/Abandon trigger unexpected errors